src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetServiceMessagePackRpcDescriptor.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using MessagePack;
+using MessagePack.Formatters;
+using MessagePack.Resolvers;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using StreamJsonRpc;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    internal class NuGetServiceMessagePackRpcDescriptor : ServiceJsonRpcDescriptor
```

Comment:
Should this be extension from ServiceJson?

Reply:
I assume you are asking about the name of the class being MessagePack and deriving from ServiceJson? This is the correct class to be deriving from.

Reply:
Yes, it just read funny :D 
Sounds like it's on purpose.